crates/torii/src/graphql/system_call.rs

```diff
@@ -1,67 +1,91 @@
-use async_graphql::{ComplexObject, Context, Result, SimpleObject};
+use async_graphql::dynamic::{Field, FieldFuture, FieldValue, InputValue, TypeRef};
+use async_graphql::Value;
 use chrono::{DateTime, Utc};
+use indexmap::IndexMap;
 use serde::Deserialize;
-use sqlx::pool::PoolConnection;
-use sqlx::{Pool, Sqlite};
+use sqlx::{FromRow, Pool, Sqlite};
 
-use super::system::{system_by_id, System};
+// use super::system::System;
+use super::{ObjectTraitInstance, ObjectTraitStatic, TypeMapping, ValueMapping};
 
-#[derive(SimpleObject, Debug, Deserialize)]
+#[derive(FromRow, Deserialize)]
 #[serde(rename_all = "camelCase")]
-#[graphql(complex)]
 pub struct SystemCall {
     pub id: i64,
-    pub system_id: String,
     pub transaction_hash: String,
-    pub data: Option<String>,
+    pub data: String,
     pub created_at: DateTime<Utc>,
+    #[serde(skip_deserializing)]
+    pub system_id: String,
+}
+pub struct SystemCallObject {
+    pub field_type_mapping: TypeMapping,
 }
 
-#[ComplexObject]
-impl SystemCall {
-    async fn system(&self, context: &Context<'_>) -> Result<System> {
-        let mut conn = context.data::<Pool<Sqlite>>()?.acquire().await?;
-        system_by_id(&mut conn, self.system_id.clone()).await
+impl ObjectTraitStatic for SystemCallObject {
+    fn new() -> Self {
+        Self {
+            field_type_mapping: IndexMap::from([
+                (String::from("id"), String::from("ID")),
+                (String::from("transactionHash"), String::from("String")),
+                (String::from("data"), String::from("String")),
+                (String::from("createdAt"), String::from("DateTime")),
```

Comment:
are these values coming from the transactions block time? created At feels a bit confusing since it is usually when the row in the db is created. maybe we should have "timestamp"?

Reply:
Nope, these are all db insertion timestamp, it's on most of the tables: https://github.com/dojoengine/dojo/blob/main/crates/torii/migrations/20230316154230_setup.sql

Should we include blocktime too?

Reply:
Gonna merge this for now and add blocktime in another PR if needed